### PR TITLE
add login and signup pages

### PR DESCRIPTION
**Pull request by Dzemite**

### CODE REVIEW

File: frontend/src/containers/LoginForm/LoginForm.module.scss [hunk 1]
```diff
@@ -1,8 +1,9 @@
 .loginFormWrapper {
   display: flex;
   flex-direction: column;
+  gap: 8px;
```
Comment (Magkoffe50) [inline frontend/src/containers/LoginForm/LoginForm.module.scss:4]: Gap is not working on <ios 15.5(latest). Try to use margin:  0-8px; for container and margin:0 8px; for item

File: frontend/src/containers/LoginForm/LoginForm.module.scss [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
   width: 20vw;
-  min-width: 150px;
+  min-width: 250px;
   padding: 32px;
   box-sizing: content-box;
```
Comment (Magkoffe50) [inline frontend/src/containers/LoginForm/LoginForm.module.scss:8]: Are you shure that it needed to use conte-box?

File: frontend/src/containers/RouterManager/index.tsx [hunk 3]
```diff
@@ -1,10 +1,15 @@
+import { RequireAuth } from 'hoc';
 import { FC } from 'react';
 import { Route, Routes } from 'react-router-dom';
 import { routes } from 'router';
 
 const RouteManager: FC = () => {
   const router = routes.map((route) => {
-    return <Route path={route.path} element={route.component} key={route.name} />;
+    return <Route path={route.path} element={
```
Comment (Magkoffe50) [inline frontend/src/containers/RouterManager/index.tsx:8]: Read about React laze and try to add component, witch will create lazy modules, if it module(page) is protectd - redirect to home page for example

File: frontend/src/index.tsx [hunk 4]
```diff
@@ -1,32 +1,25 @@
...
 const container = document.getElementById('root');
-// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 const root = createRoot(container!);
-const queryClient = new QueryClient()
+const store = setupStore();
+
 root.render(
-  <QueryClientProvider client={queryClient}>
-    <Provider store={store.store}>
-      <PersistGate loading={null} persistor={store.persistor}>
```
Comment (Magkoffe50) [inline frontend/src/index.tsx:22]: It helps you simply save at local storage data you need(when it rewrite on readux soter), and in your actionCreator you may delete saving to ls code.

File: frontend/src/store/reducers/SignupReducer/ActionCreator.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        email: credentials.email,
+        password: credentials.password,
+      }}
+    );
+    
+    if ((response as {error: ErrorResponse | SerializedError;}).error) {
+      return thunkAPI.rejectWithValue('Something went wrong.');
+    }
+
+    localStorage.setItem('token', (response as {data: SignupMutation;}).data?.register.jwt);
```
Comment (Magkoffe50) [inline frontend/src/store/reducers/SignupReducer/ActionCreator.ts:32]: PersistGate component help you with saving on ls

**Review by Magkoffe50 — APPROVED** [review]

File: frontend/src/containers/LoginForm/LoginForm.module.scss (hunk 2 above)
Comment (Dzemite) [reply frontend/src/containers/LoginForm/LoginForm.module.scss:8]: Yeap

**Pull request merged**